File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool fun(string &s, int l, int r){
        int n=s.length();
        while(l<n && !isalnum(s[l])) l++;
        while(r>=0 && !isalnum(s[r])) r--;
        if(l>=r) return 1;
        if(tolower(s[l])!=tolower(s[r])) return 0;
        return fun(s,l+1,r-1);
    }
    bool isPalindrome(string s) {
        return fun(s,0,s.length()-1);
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        n=len(s)
        l=0
        r=n-1
        while(l<r):
            while( l<n and s[l].isalnum() == 0):
                l+=1
            while(r>-1 and s[r].isalnum() == 0):
                r-=1
            if(l<n and r>-1 and s[l].lower() != s[r].lower()):
                return False
            l+=1
            r-=1
        return True
        